addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParser.java

```diff
@@ -19,49 +19,108 @@
  */
 package org.zaproxy.addon.graphql;
 
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import graphql.introspection.IntrospectionQuery;
+import graphql.introspection.IntrospectionResultToSchema;
+import graphql.language.Document;
+import graphql.schema.idl.SchemaPrinter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Map;
 import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.io.FileUtils;
 import org.apache.log4j.Logger;
-import org.parosproxy.paros.control.Control;
-import org.parosproxy.paros.extension.history.ExtensionHistory;
-import org.parosproxy.paros.model.HistoryReference;
-import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.Constant;
 import org.parosproxy.paros.network.HttpMessage;
-import org.parosproxy.paros.network.HttpSender;
-import org.zaproxy.zap.utils.ThreadUtils;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.network.HttpRequestBody;
 
 public class GraphQlParser {
 
     private static final Logger LOG = Logger.getLogger(GraphQlParser.class);
+    private final URI endpointUrl;
+    private final int initiator;
+    private final RequestHandler requestHandler;
+    private HttpMessage importMessage;
```

Comment:
It should not be an instance variable, the persistence should be done through the `RequestHandler` automatically when the message is sent.

Reply:
But you had also suggested earlier to have a separate method for persisting the message, so when we write tests we can skip persisting the messages. Is it okay if I modify the RequestHandler#makeRequest method to also accept a boolean `persist` as a parameter?

Reply:
Sure, but we should handle them immediately not keep references to it. It would be better to allow the user to choose how it's handled (callback) but a boolean will do.